src/keyboard/k884x.rs

```diff
@@ -0,0 +1,91 @@
+use anyhow::{ensure, Result};
+use log::debug;
+use rusb::{Context, DeviceHandle};
+
+use super::{Key, Keyboard, Macro, MouseAction, MouseEvent};
+
+pub struct Keyboard884x {
+    handle: DeviceHandle<Context>,
+    endpoint: u8,
+}
+
+impl Keyboard for Keyboard884x {
+    fn bind_key(&mut self, layer: u8, key: Key, expansion: &Macro) -> Result<()> {
+        ensure!(layer <= 15, "invalid layer index");
+
+        debug!("bind {} on layer {} to {}", key, layer, expansion);
+
+        let mut msg = vec![
```

Comment:
`Vec::with_capacity`

Reply:
Not sure I understand, could you give an example of what you suggest exactly?